src/config/from_document.rs

```diff
@@ -273,14 +273,14 @@ fn to_inline(directives: &[Positioned<ConstDirective>]) -> Option<config::Inline
     }
   })
 }
-fn to_http(directives: &[Positioned<ConstDirective>]) -> Option<config::Http> {
-  directives.iter().find_map(|directive| {
+fn to_http(directives: &[Positioned<ConstDirective>]) -> Valid<Option<config::Http>> {
+  let mut http = Valid::Ok(None);
+  for directive in directives {
     if directive.node.name.node == "http" {
-      config::Http::from_directive(&directive.node).ok()
-    } else {
-      None
+      http = Http::from_directive(&directive.node).map(Some)
```

Comment:
It might be not a big deal, but using for in this way will iterate over all the values despite if http has been found at the start. While previous code with `find_map` will stop on first entry and return the value.

To mitigate this we could just use return here on first match